### DISCUSSION TITLE
How to get the same behavior as ios exactly?

### DISCUSSION THREAD

**Discussion by wcastand** [zoontek/react-native-edge-to-edge#37] [Q&A] [answered]
in my ios app currently i get the status bar but the content flow behind it, while the status bar on android is a hard white and the lib as far as i know doesn't let me change that.

is it possible to get the status bar translucent and let the content flow behind like the description says on the readme?

or i missunderstood or did something wrong. i'm on expo and added the lib to the `plugins`

and in my code `<SystemBars style="dark" hidden={{ statusBar: false, navigationBar: true }} />`

i don't want to hide the status bar since most phone these days have a knob at the top anyway so can't put content there but i'd like a translucent status bar with a consistent padding/margin/safe area between both platform. maybe it's not the goal of this lib? 

did i misunderstood that part of the readme `allowing your Android app content to flow seamlessly beneath the system bars.` ?

![Screenshot 2024-10-23 at 12 38 28](https://github.com/user-attachments/assets/e2baaa13-d805-4025-ab64-8a51c00f0c13)

**Comment by zoontek** [comment]
You might got something wrong on how Expo works. Did you created a build?

![image](https://github.com/user-attachments/assets/6d8d50de-ca6c-4f3a-a299-a6ad01826cfa)

You need to do a prebuild and run the app using `npx expo run:android`

**Reply by wcastand** [reply]
i did a prebuild, my app is a expo-dev app already so i added to app config, prebuild and rebuild android and ios apps. removed expo-status-bar and navbar too to avoid conflicts.

**Reply by zoontek** [reply, marked as answer]
@wcastand Just in case, are you using a library that also perform (kinda) edge-to-edge on its side?

Libraries like `react-native-keyboard-controller` have `statusBarTranslucent` and `navigationBarTranslucent` on `KeyboardProvider`. If those are not set to `true`, the library will re-enable opaque system bars.
In the future, they are expect to adopt the small helper `react-native-is-edge-to-edge` to avoid this kind of conflicts.

Publish your `package.json`, it will be easy to check.

**Reply by wcastand** [reply]
oh we have those exact, i will check again, pretty sure it's something in our setup, tried on a new repo (npx create-expo-app) and it works as expected.

most likely the deps like you said. thanks for the help 🙏

**Reply by zoontek** [reply]
Moving an entire ecosystem is not an easy task, but we will get there!

**Reply by wcastand** [reply]
so been trying to figure it out, i set the props of `react-native-keyboard-controller` to false to be sure. i put the lib in the `plugins` list at the end to be applied last (i also tried first).

i tried to check my layout, see if any was setting white color or something but can't figure it out. i keep having the white status bar at the top, i try removing safe-area too but nothing seems to be causing the white bar. if i can figure it out i will post it here in case anyone has the same problem.

**Reply by zoontek** [reply]
> I set the props of react-native-keyboard-controller to false to be sure

They have to be set to `true`

**Reply by wcastand** [reply]
🤯 ok working now so it was `react-native-keyboard-controller` needing the props set to true, i misunderstood what i was suppose to do. my bad thanks for the help and swift answers !

**Reply by zoontek** [reply]
@wcastand Soon, the default value for those props will be `statusBarTranslucent = isEdgeToEdge()` instead of `false`, and people will stop bother about this 🔥

**Reply by wcastand** [reply]
you can add `@lodev09/react-native-true-sheet` in the deps that doesn't work properly with edge-to-edge. when i open a sheet, it shows the nav bar again. probably the same issue as the `Modal` you described. will look with the maintainer what can be done.

**Reply by zoontek** [reply]
It's an issue opened on their side, yes ([link](https://github.com/lodev09/react-native-true-sheet/issues/72))
